stl/inc/xstring

```diff
@@ -2642,6 +2642,44 @@ private:
     }
 
 public:
+#if _HAS_CXX20
+    void _Move_assign_from_buffer(_Elem* const _Right, const size_type _Size, const size_type _Res) {
+        // Move assign from a buffer, used by basic_stringbuf
+        _Tidy_deallocate();
+        _Right[_Size]  = _Elem();
+        auto& _My_data = _Mypair._Myval2;
+        _Construct_in_place(_My_data._Bx._Ptr, pointer_traits<pointer>::pointer_to(*_Right));
+        _My_data._Mysize = _Size;
+        _My_data._Myres  = _Res;
+    }
+
+    struct _Released_buffer {
+        _Elem* _Ptr;
+        size_type _Size;
+        size_type _Res;
+    };
+
+    _NODISCARD _Released_buffer _Release_to_buffer() {
+        // Release to a buffer, or allocate a new one if in small string mode
+        _Released_buffer _Result;
+        auto& _My_data = _Mypair._Myval2;
+        if (_My_data._Large_string_engaged()) {
+            _Result._Ptr      = _Unfancy(_My_data._Bx._Ptr);
+            _My_data._Bx._Ptr = nullptr;
+            _Result._Size     = _STD exchange(_My_data._Mysize, size_type{0});
+            _Result._Res      = _STD exchange(_My_data._Myres, size_type{0});
+            _Tidy_init();
+        } else {
+            _Result._Ptr = _Unfancy(_Getal().allocate(_BUF_SIZE));
+            _Traits::copy(_Result._Ptr, _My_data._Bx._Buf, _BUF_SIZE);
+            _Result._Size = _My_data._Mysize;
+            _Result._Res  = _BUF_SIZE;
+            _Tidy_deallocate();

```

Comment:
No reason to _Tidy_deallocate here -- we know there's nothing to deallocate because the string is in 'small' mode. Maybe just leave the string contents alone?

Reply:
Maybe using `_Tidy_init` is better. According to the standard, the string is moved away, so it should be tidy now.

Reply:
A moved from string is not necessarily empty; and here it would take more work to make it empty, not less work.

Reply:
You're right.